### PR TITLE
Alert lights for SSU's with mobs inside

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Adds flashing lights for SSUs with mobs in them.

## Why It's Good For The Game
Oh lord there's a dead guy in there!

## Images of changes
NOTE: IMAGE IS OLD, SPEED OF FLASHING IS SLOWED DOWN 5X IN NEW VERSION

https://user-images.githubusercontent.com/12197162/174441748-4136782d-ca97-4d27-b827-235b582bd37b.mp4

## Changelog
:cl:
add: Added flashy lights for suit storage units that contain mobs
/:cl:

### CODE REVIEW

**Comment by Bm0n** [comment]
Wait did the old ones do this? Because if not that's quite a unique way for an antag to hide a kill target without blatantly round removing them. This kind of ruins that.

**Comment by S34NW** [comment]
> 
> Wait did the old ones do this? Because if not that's quite a unique way for an antag to hide a kill target without blatantly round removing them. This kind of ruins that.

The old ones had a tell yes, they were slightly transparent and put a naked human mob sprite behind the SSU when someone was inside, so you could see there was a person in there.

File: code/game/machinery/suit_storage_unit.dm [hunk 1]
```diff
@@ -309,24 +311,34 @@
...
+		add_overlay(alert_lights)
 		return
 
 	if(state_open)
 		add_overlay("[base_icon_state]_open")
-		add_overlay("[base_icon_state]_ready")
+		alert_lights = icon(icon, "[base_icon_state]_ready")
+		if(occupant)
+			alert_lights = icon(icon, "[base_icon_state]_body")
+		add_overlay(alert_lights)
```
Comment (SteelSlayer) [inline code/game/machinery/suit_storage_unit.dm:329]: I don't like the new way this is being handled with the `alert_lights` var. The `icon()` proc `new`s an icon every time it's called. But, it's possible that that icon is not even used, and another one is `new`'d further down the chain.

I suggest doing away with `alert_lights` and either just adding the overlay or not.

File: code/game/machinery/suit_storage_unit.dm [hunk 2]
```diff
@@ -309,25 +311,25 @@
...
 			add_overlay("[base_icon_state]_helm")
 		if(storage)
 			add_overlay("[base_icon_state]_storage")
-		return
-	add_overlay("[base_icon_state]_ready")
-	add_overlay("[base_icon_state]_lights_closed")
+	else
+		add_overlay("[base_icon_state]_lights_closed")
+
+	add_overlay(icon(icon, "[base_icon_state]_[occupant ? "body" : "ready"]"))
```
Comment (SteelSlayer) [inline code/game/machinery/suit_storage_unit.dm:332]:
```suggestion
	add_overlay("[base_icon_state]_[occupant ? "body" : "ready"]"))
```

File: code/game/machinery/suit_storage_unit.dm [hunk 3]
```diff
@@ -309,25 +311,25 @@
 
 	if(uv)
 		if(uv_super)
-			add_overlay("[base_icon_state]_uvstrong")
 			add_overlay("[base_icon_state]_super")
+			add_overlay(icon(icon, "[base_icon_state]_[occupant ? "body" : "uvstrong"]"))
 		else
-			add_overlay("[base_icon_state]_uv")
+			add_overlay(icon(icon, "[base_icon_state]_[occupant ? "body" : "uv"]"))
```
Comment (SteelSlayer) [inline code/game/machinery/suit_storage_unit.dm:317]:
```suggestion
			add_overlay("[base_icon_state]_[occupant ? "body" : "uv"]"))
```

File: code/game/machinery/suit_storage_unit.dm [hunk 4]
```diff
@@ -309,25 +311,25 @@
 
 	if(uv)
 		if(uv_super)
-			add_overlay("[base_icon_state]_uvstrong")
 			add_overlay("[base_icon_state]_super")
+			add_overlay(icon(icon, "[base_icon_state]_[occupant ? "body" : "uvstrong"]"))
```
Comment (SteelSlayer) [inline code/game/machinery/suit_storage_unit.dm:315]:
```suggestion
			add_overlay("[base_icon_state]_[occupant ? "body" : "uvstrong"]"))
```

File: code/game/machinery/suit_storage_unit.dm [hunk 5]
```diff
@@ -48,6 +48,8 @@
 	/// If set, turned into typecache in Initialize, other wise, defaults to mob/living typecache
 	var/list/occupant_typecache
 	var/atom/movable/occupant = null
+	///Tracks alert light overlay for easy light implementations
+	var/alert_lights
```
Comment (SteelSlayer) [inline code/game/machinery/suit_storage_unit.dm:52]: Unused now.
```suggestion
```

File: code/game/machinery/suit_storage_unit.dm [hunk 6]
```diff
@@ -309,25 +309,25 @@
 
 	if(uv)
 		if(uv_super)
-			add_overlay("[base_icon_state]_uvstrong")
 			add_overlay("[base_icon_state]_super")
+			add_overlay("[base_icon_state]_[occupant ? "body" : "uvstrong"]"))
```
Comment (S34NW) [inline code/game/machinery/suit_storage_unit.dm:313]:
```suggestion
			add_overlay("[base_icon_state]_[occupant ? "body" : "uvstrong"]")
```

File: code/game/machinery/suit_storage_unit.dm [hunk 7]
```diff
@@ -309,25 +309,25 @@
...
 		else
-			add_overlay("[base_icon_state]_uv")
+			add_overlay("[base_icon_state]_[occupant ? "body" : "uv"]"))
```
Comment (S34NW) [inline code/game/machinery/suit_storage_unit.dm:315]:
```suggestion
			add_overlay("[base_icon_state]_[occupant ? "body" : "uv"]")
```

File: code/game/machinery/suit_storage_unit.dm [hunk 8]
```diff
@@ -309,25 +309,25 @@
...
 			add_overlay("[base_icon_state]_helm")
 		if(storage)
 			add_overlay("[base_icon_state]_storage")
-		return
-	add_overlay("[base_icon_state]_ready")
-	add_overlay("[base_icon_state]_lights_closed")
+	else
+		add_overlay("[base_icon_state]_lights_closed")
+
+	add_overlay("[base_icon_state]_[occupant ? "body" : "ready"]"))
```
Comment (S34NW) [inline code/game/machinery/suit_storage_unit.dm:330]:
```suggestion
	add_overlay("[base_icon_state]_[occupant ? "body" : "ready"]")
```

**Review by S34NW — COMMENTED** [review]
welp

**Review by SteelSlayer — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**